custom_components/sleep_as_android/sensor.py

```diff
@@ -101,11 +132,11 @@ async def async_added_to_hass(self):
         _LOGGER.debug("My device id is %s", device.id)
         self._device_id = device.id
 
-        if (old_state := await self.async_get_last_state()) is not None:
-            self._attr_native_value = old_state.state
+        if (old_state := await self.async_get_last_sensor_data()) is not None:
+            self._attr_native_value = old_state.native_value
```

Comment:
Why `native_value`, but not `state` here and below?

Looks like `state` is the final state of sensor from HA point of view: HA convert timestamps, date and temperature types, but keep all others, so we may keep `state` here and below.

Reply:
I should test this out what the best way is

Reply:
> I should test this out what the best way is

`native_value` is looks fine too (now): we are setting native value, so compare with old native value is OK.
We may keep it. Sorry, I was not careful enough.

Reply:
Haha no worries.

I forgot to update the PR tho. Will do that probably tomorrow